File: Kurs C++/Zad6/podziel.cpp
//
// Created by Mateusz Pater on 08.04.2016.
//

#include "podziel.h"


double podziel::oblicz() {

    return a->oblicz()/b->oblicz();
}
podziel::podziel(wyrazenie *a, wyrazenie *b){
    this->a = a;
    this->b = b;
    if( b->oblicz() == 0 ) {  throw std::invalid_argument("Dzielenie przez 0"); }
}

std::string podziel::formatuj_wyrazenie() {
    return "(" + a->formatuj_wyrazenie() + " / " + b->formatuj_wyrazenie() + ")";
}

File: Kurs Java/Zadanie 3/src/com/company/Podziel.java
package com.company;

/**
 * author @pater
 */
public class Podziel extends Operator2Arg{

    public Podziel(Wyrazenie a, Wyrazenie b) {
        super(a,b);

    }

    @Override
    public double oblicz() {

        if(b.oblicz() == 0)
        {
            throw new IllegalArgumentException("Cholero, ie dziel przez 0:>");
        }

        return a.oblicz() / b.oblicz();
    }
    @Override
    public String toString() {
        return  a + "/" + b;
    }
}
